default/AI/ShipDesignAI.py

```diff
@@ -1916,7 +1925,13 @@ def _get_design_by_name(design_name, verbose=False):
     :param design_name: string
     :return: shipDesign object
     """
-    if design_name in Cache.design_id_by_name:
+    # get shipdesign from fo if
+    #  * not in cache or
+    #  * if an design is invalid and update_invalid is true
+    # otherwise use cache
+    if design_name in Cache.design_id_by_name and (
+            ( not update_invalid ) or
+            ( Cache.design_id_by_name[design_name] != EnumsAI.AIShipDesignTypes.SHIPDESIGN_INVALID ) ) :
```

Comment:
Instead of ```(not A) or (x!=y)``` ==> ```not (A and x==y)``` seems much easier to read.

Also no whitespace immediately inside of parantheses: ```( foo )``` ==> ```(foo)```
While some of the older code is a mess, we agreed to stick to PEP8-conventions for new code / cleaning up the old one. (https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements)